yt_dlp/extractor/acfun.py

```diff
@@ -0,0 +1,191 @@
+            video_list = json_all['videoList']
+            p_idx, p_video_info = next(
+                (idx + 1, v) for (idx, v) in enumerate(video_list)
+                if v['id'] == video_internal_id)
+
+            if len(video_list) > 1:
+                title = f'{title} P{p_idx:02d} {p_video_info["title"]}'
+
+        return {
+            'id': video_id,
+            'title': title,
+            'thumbnail': json_all.get('coverUrl'),
+            'description': json_all.get('description'),
+            'uploader': traverse_obj(json_all, ('user', 'name')),
+            'uploader_id': traverse_obj(json_all, ('user', 'href')),
+            'tags': traverse_obj(json_all, ('tagList', ..., 'name')),
+            'view_count': int_or_none(json_all.get('viewCount')),
+            'like_count': int_or_none(json_all.get('likeCountShow')),
+            'comment_count': int_or_none(json_all.get('commentCountShow')),
+            'http_headers': {
+                'Referer': url,
+            },
+            **self.parse_format_and_subtitle(video_id, video_info)
+        }
+
+
+class AcFunBangumiIE(AcFunVideoBaseIE):
+    _VALID_URL = r'(?x)https?://(?:www\.acfun\.cn/bangumi/)(?P<id>aa[_\d]+(?:\?ac=\d+)?)'
+
+    _TESTS = [{
+        'url': 'https://www.acfun.cn/bangumi/aa6002917',
+        'info_dict': {
+            'id': 'aa6002917',
+            'title': '租借女友 第1话 租借女友',
+            'duration': 1467,
+            'timestamp': 1594432800,
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': 'm3u8',
+        },
+        'skip': 'Geo-restricted to China',
+    }, {
+        'url': 'https://www.acfun.cn/bangumi/aa6002917_36188_1745457?ac=2',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        base_id, ac_idx = self._search_regex(r'(?P<id>aa[_\d]+)(?:\?ac=(?P<ac_idx>\d+))?', video_id,
+                                             'ac_idx_parse', group=['id', 'ac_idx']) or (None, None)
+        if base_id is not None:
+            video_id = f'{base_id}{format_field(ac_idx, template="_%s")}'
+
+        webpage = self._download_webpage(url, video_id)
+        json_bangumi_data = self._search_json(r'window.bangumiData\s*=\s*', webpage, 'bangumiData', video_id)
+
+        info = {}
+        if not ac_idx:
+            if 'currentVideoInfo' not in json_bangumi_data:
+                raise ExtractorError(f'Unknown webpage json schema{bug_reports_message()}')
+            video_info = json_bangumi_data['currentVideoInfo']
+            title = json_bangumi_data.get('showTitle', '')
+            season_id = json_bangumi_data.get('bangumiId')
+
+            season_number = None
+            if season_id:
+                season_number = next((
+                    idx + 1 for (idx, v) in enumerate(json_bangumi_data.get('relatedBangumis', []))
+                    if v.get('id') == season_id), 1)
```

Comment:
You sure the default should be 1 and not `None`?

Reply:
I think so, if it is the first sesaon, relatedBangumis is empty